Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp

```diff
@@ -1796,8 +1796,9 @@ ObjectContentType WebFrameLoaderClient::objectContentType(const URL& url, const
         return ObjectContentType::Frame;
 
 #if PLATFORM(IOS_FAMILY)
-    // iOS can render PDF in <object>/<embed> via PDFDocumentImage.
-    if (MIMETypeRegistry::isPDFOrPostScriptMIMEType(mimeType))
+    // iOS can render PDF in <object>/<embed> via PDFDocumentImage, except if it is
+    // via a data URL.
+    if (MIMETypeRegistry::isPDFOrPostScriptMIMEType(mimeType) && !url.protocolIsData())
```

Comment:
This strikes me as odd, but if we're just hardening the existing truth, it's OK. It would be nice to understand *why* data: doesn't work here, but if that's not what recently broke, I'm on board.

Reply:
I'll investigate why the data: case is failing. I didn't focus on this because it wasn't what regressed but I think it'd be nice if we could make it work.

Reply:
Hmm, it is a little more complicated than I thought. We do support data URLs here. The issue is the actual data URL used by the site. They use "data:application/pdf;base64,aG1t" (like in my test). "aG1t" in base64 is decoded to "hmm". Then we pass "hmm" as data to PDFKit when constructing the PDFDocument. This is invalid so [PDFDocument initWithData:] returns nil and this causes us to fail the "image" load and dispatch an error event.

Given those findings, I am not sure how to deal with this.